common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -1159,7 +1159,13 @@ def create_child(self, user_id, parent_usage_key, block_type, block_id=None, **k
         # attach to parent if given
         if 'detached' not in xblock._class_tags:
             parent = self.get_item(parent_usage_key)
-            parent.children.append(xblock.location)
+
+            # Originally added to support entrance exams (settings.FEATURES.get('ENTRANCE_EXAMS'))
+            if kwargs.get('position') is None:
+                parent.children.append(xblock.location)
+            else:
+                parent.children.insert(kwargs.get('position'), xblock.scope_ids.usage_id)
+
```

Comment:
why would one add an xblock.location and the other xblock.scope_ids.usage_id. I'm guessing they are equivalent, but - if so - then I'd suggest unifying both cases here to make it more readible